listenbrainz/webserver/static/js/src/types.d.ts

```diff
@@ -2,8 +2,43 @@ declare module "spotify-web-playback-sdk";
 declare module "react-bs-notifier";
 declare module "time-ago";
 
-// TODO: make this type specific
-declare type Listen = any;
+// TODO: Remove "| null" when backend stops sending fields with null
+declare type Listen = {
+  listened_at: number;
+  listened_at_iso?: string | null;
```

Comment:
Is there a way to get rid of these `| null`s, they seem non-ideal.

Reply:
We can turn off `strictNullChecks` in typescript which will treat `null` and `undefined` as the same everywhere. But then we will also lose other errors such as **Possibly null/undefined**.

Reply:
That does not sound like a good idea. Let's keep the nulls for now.